shared/util/saga.js

```diff
@@ -0,0 +1,11 @@
+// @flow
+
+import {put, cps} from 'redux-saga/effects'
+
+export function * cpsWithWaiting (waitingFn: (isWaiting: boolean) => any, cpsFn: any, ...args: any): any {
+  yield put(waitingFn(true))
+  const value = yield cps(cpsFn, ...args)
```

Comment:
I used cps because the rpc is already expecting a callback of the same form, and it makes it a little smaller.

A promise could have worked just as well